### ISSUE TITLE
`require-returns-check`: false-positive with break in switch in other switch

### ISSUE THREAD

**Issue by regseb** [gajus/eslint-plugin-jsdoc#923]
## Expected behavior

If a `switch` contains another `switch` with a `break`, the `require-returns-check` rule shouldn't report an error.

## Actual behavior

If a `switch` contains another `switch` with a `break`, the `require-returns-check` rule report an error.

## ESLint Config (in package.json)

```json
{
    "name": "testcase",
    "version": "1.0.0",
    "private": true,
    "dependencies": {
        "eslint": "8.26.0",
        "eslint-plugin-jsdoc": "39.3.17"
    },
    "eslintConfig": {
        "plugins": ["jsdoc"],
        "rules": {
            "jsdoc/require-returns-check": 2
        }
    }
}
```

## ESLint sample

```js
/**
 * Foo.
 *
 * @returns Bool.
 */
function foo(bar) {
    switch (bar[0]) {
        case "baz":
            switch (bar[1]) {
                case "qux": break;
            }
            return true;
        default:
            return false;
    }
};
```

## Environment

- Node version: v18.11.0
- ESLint version  8.26.0
- `eslint-plugin-jsdoc` version: 39.3.17

## Additional info

The problem started in version [39.3.15](https://github.com/gajus/eslint-plugin-jsdoc/compare/v39.3.14...v39.3.15).
Maybe the `require-returns-check` rule thinks that the `break` applies to the highest `switch`.

**Comment by brettz9** [comment]
Thanks for the report. This should be fixed now, but just be aware the code for checking all branches is fresh and is somewhat involved, so there may be some other areas needing improvement.
